presto-main/src/main/java/io/prestosql/type/TypeUtils.java

```diff
@@ -45,6 +47,15 @@ public static int expectedValueSize(Type type, int defaultSize)
         if (type instanceof FixedWidthType) {
             return ((FixedWidthType) type).getFixedSize();
         }
+        else if (type instanceof VarcharType) {
+            // If bound on length of varchar is smaller than defaultSize, use that as expected size
+            return ((VarcharType) type).getLength()
+                    .map(length -> Math.min(length, defaultSize))
+                    .orElse(defaultSize);
+        }
+        else if (type instanceof CharType) {
```

Comment:
Chart doesn't occupy full declared length. Trailing spaces are not stored.

Reply:
Currently CharType is not allowed to store trailing spaces https://github.com/prestosql/presto/blob/283f5b3c42002082dcd68b8eab37432b5b90b32f/presto-spi/src/main/java/io/prestosql/spi/type/CharType.java#L144
Though you're right about not occupying full declared length, will change this.